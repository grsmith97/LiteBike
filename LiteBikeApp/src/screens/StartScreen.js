import React from 'react';
import {StyleSheet, Text, Button, View} from 'react-native';

const StartScreen = ({navigation}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title} data-testing='litebike-title'>
        LiteBike
      </Text>
      <Button
        title='Start'
        data-testing='start-button'
        color='#add8e6'
        style={styles.button}
        onPress={() => navigation.navigate('EndScreen')}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    color: '#1344ad',
    fontSize: 40,
  },
  button: {
    // backgroundColor: '#1344ad',
  }
});

export default StartScreen;
