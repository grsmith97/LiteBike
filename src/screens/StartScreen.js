import React from 'react';
import {StyleSheet, Text, Button, View, TouchableOpacity, Image} from 'react-native';
import settings from '../assets/settings-cog.png';

const StartScreen = ({navigation}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title} data-testing='litebike-title'>
        LiteBike
      </Text>
      <Button
        title='Start'
        data-testing='start-button'
        color='#add8e6'
        onPress={() => navigation.navigate('MapScreen')}
      />
      <TouchableOpacity disabled={true} data-testing='settings-disabled'>
        <Image style={styles.settings} source={settings} data-testing='settings-button' />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    color: '#1344ad',
    fontSize: 40,
  },
  settings: {
    height: 40,
    width: 40,
  }
});

export default StartScreen;
