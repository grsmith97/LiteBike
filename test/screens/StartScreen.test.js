import React from 'react';
import renderer, {act} from 'react-test-renderer';
import StartScreen from '../../src/screens/StartScreen';
import settingsImage from '../../src/assets/settings-cog.png';

let component, settingsButton, settingsDisabled, startButton, text;

describe('Given StartScreen component', () => {
  describe('When the component renders with a message', () => {
    const props = {
      navigation: {
        navigate: jest.fn()
      }
    };

    beforeEach(() => {
      component = renderer.create(<StartScreen {...props}/>);
      startButton = component.root.findByProps({'data-testing': 'start-button'});
      text = component.root.findByProps({'data-testing': 'litebike-title'});
      settingsButton = component.root.findByProps({'data-testing': 'settings-button'});
      settingsDisabled = component.root.findByProps({'data-testing': 'settings-disabled'});
    });

    it('Then the component matches the snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('Then LiteBike appears on the screen', () => {
      expect(text.props.children).toBe('LiteBike');
    });

    it('Then the start button is blue', () => {
      expect(startButton.props.color).toBe('#add8e6');
    });

    it('Then the settings button exists', () => {
      expect(settingsButton.props.source).toBe(settingsImage);
    });

    it('Then the settings button is disabled', () => {
      expect(settingsDisabled.props.disabled).toBe(true);
    });

    describe('When the start button is clicked', () => {
      beforeEach(() => {
        act(() => {
          startButton.props.onPress();
        });
      });

      it('Then navigate to the map screen', () => {
        expect(props.navigation.navigate).toHaveBeenCalled();
      });
    });
  });
});
