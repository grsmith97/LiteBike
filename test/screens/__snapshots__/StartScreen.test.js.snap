// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given StartScreen component When the component renders with a message Then the component matches the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    data-testing="litebike-title"
    style={
      Object {
        "color": "#1344ad",
        "fontSize": 40,
      }
    }
  >
    LiteBike
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#add8e6",
            },
          ]
        }
      >
        Start
      </Text>
    </View>
  </View>
</View>
`;
